libraries/AP_Common/AP_FWVersionDefine.h

```diff
@@ -23,10 +23,23 @@
 #include <AP_Common/AP_FWVersion.h>
 
 const AP_FWVersion AP_FWVersion::fwver{
+    // Version header struct
+    .header = 0x61706677766572fb, // First 7 MSBs: "apfwver", LSB is the checksum of the previous string: 0xfb
+    .header_version = static_cast<uint16_t>(0x0100), // Major and minor version
+    .pointer_size = sizeof(void*),
+    .vehicle_type = static_cast<uint8_t>(APM_BUILD_DIRECTORY)
```

Comment:
This is a little weird.

Reply:
Just to make sure, are you talking about the cast or `APM_BUILD_DIRECTORY` definition ? For the definition, check my question here: https://discord.com/channels/674039678562861068/674039678982422579/758428712093220904

Reply:
Sorry, it just looks like you're casting a string into a `uint8_t` here...